### PR TITLE
{Matlab} create muscle curve script

### PR DESCRIPTION
**Pull request by jimmyDunne**
Plots active and passive force-length curves of a model.

### CODE REVIEW

File: OpenSim/Sandbox/MatlabScripts/muscleCurves.m [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+            % Store all the data in the zero matrix
+            storageData(j,:) = [...
+                rad2deg(coordValue) ...                        % Coordinate Value
+                myMuscle.getFiberLength(s) ...             % Fiber length
+                myMuscle.getNormalizedFiberLength(s) ...   % Normalized Fibre Length
+                myMuscle.getActiveFiberForce(s) ...        % Active Force
+                myMuscle.getPassiveFiberForce(s) ];        % passive fibre forces
+
+                % check to see if that the fiber length has already been se
+            if isempty( find( myMuscle.getNormalizedFiberLength(s) == flMatrix(:,1), 1 ) )
```
Comment (tkuchida) [inline OpenSim/Sandbox/MatlabScripts/muscleCurves.m]: Not clear why this check is necessary, and the comment did not help.

File: OpenSim/Sandbox/MatlabScripts/muscleCurves.m [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+
+
+    % Reset the coordinate value back to zero
+    updCoord.setValue(s, 0);
+end
+
+fl_active = flMatrix(:,1:2);
+fl_passive = flMatrix(:,[1 3]);
+
+fl_active(find(fl_active(:,1) == 0 ),: ) = [];
```
Comment (tkuchida) [inline OpenSim/Sandbox/MatlabScripts/muscleCurves.m]: Not clear why this should be necessary. Provide a comment?

File: OpenSim/Sandbox/MatlabScripts/muscleCurves.m [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+            storageData(j,:) = [...
+                rad2deg(coordValue) ...                        % Coordinate Value
+                myMuscle.getFiberLength(s) ...             % Fiber length
+                myMuscle.getNormalizedFiberLength(s) ...   % Normalized Fibre Length
+                myMuscle.getActiveFiberForce(s) ...        % Active Force
+                myMuscle.getPassiveFiberForce(s) ];        % passive fibre forces
+
+                % check to see if that the fiber length has already been se
+            if isempty( find( myMuscle.getNormalizedFiberLength(s) == flMatrix(:,1), 1 ) )
+                flMatrix = [flMatrix ; storageData(j,3:5)];
```
Comment (tkuchida) [inline OpenSim/Sandbox/MatlabScripts/muscleCurves.m]: If only columns 3, 4, and 5 are appended to `flMatrix`, perhaps there is no need to store columns 1 (coordinate value) or 2 (fiber length).

File: OpenSim/Sandbox/MatlabScripts/muscleCurves.m [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+            myMuscle.setDefaultFiberLength( 0.01 )
+            myMuscle.setFiberLength( s, myMuscle.getOptimalFiberLength )
+            % Equilibrate the forces from the activation
+            model.equilibrateMuscles( s );
+
+            % Store all the data in the zero matrix
+            storageData(j,:) = [...
+                rad2deg(coordValue) ...                        % Coordinate Value
+                myMuscle.getFiberLength(s) ...             % Fiber length
+                myMuscle.getNormalizedFiberLength(s) ...   % Normalized Fibre Length
```
Comment (tkuchida) [inline OpenSim/Sandbox/MatlabScripts/muscleCurves.m]: "fibre" -> "fiber" to match L242 and method names? Also appears elsewhere.

**Comment by jenhicks** [comment]
@aseth1 Can you comment on how lists would be used in this script?

**Comment by jimmyDunne** [comment]
@tkuchida all your comments were addressed. However, since you wanted the function name changed, the file name had to be also changed, so your comments don't look like they have been addressed.

**Comment by tkuchida** [comment]
> the file name had to be also changed, so your comments don't look like they have been addressed.

I think my comments will disappear/collapse once you delete the obsolete file.

**Comment by tkuchida** [comment]
@jimmyDunne I've fixed a couple bugs and made some minor improvements, but I don't know what level of polish is necessary—particularly since this script is destined for the sandbox. I'll let @aseth1 and @aymanhab take it from here. :horse:

**Comment by jimmyDunne** [comment]
Thanks :)

**Pull request merged**